app/models/site_page.rb

```diff
@@ -176,6 +176,11 @@ def attributes
     attrs
   end
 
+  def header_login_enabled?
+    flag = site.site_settings.try(:header_login_enabled, site_id).try(:value)
```

Comment:
This line is calling to the `header_login_enabled` that you have defined on `SiteSetting` model. So we can change it by this:
```
SiteSetting.header_login_enabled(site_id)&.value
```